src/SubscriptionManager.ts

```diff
@@ -2,30 +2,55 @@ import { NetworkNode } from 'streamr-network'
 import { Todo } from './types'
 
 export class SubscriptionManager {
-
+    publishSessionTimeout: number
     networkNode: NetworkNode
     streams: Map<Todo,Todo>
+    publishTimeouts: Map<string, NodeJS.Timeout>
 
-    constructor(networkNode: NetworkNode) {
+    constructor(networkNode: NetworkNode, publishSessionTimeout = 30 * 1000) {
         this.networkNode = networkNode
+        this.publishSessionTimeout = publishSessionTimeout
         this.streams = new Map()
+        this.publishTimeouts = new Map()
+    }
+
+    recordPublish(streamId: string, streamPartition = 0): void {
+        this.networkNode.subscribe(streamId, streamPartition)
+
+        const key = `${streamId}::${streamPartition}`
+        clearTimeout(this.publishTimeouts.get(key)!)
+        this.publishTimeouts.set(key, setTimeout(() => {
+            this.publishTimeouts.delete(key)
+            if (!this.streams.has(key)) {
+                this.networkNode.unsubscribe(streamId, streamPartition)
+            }
+        }, this.publishSessionTimeout))
```

Comment:
I wonder if we should do this same timeout thing for subscriptions too. We have a defined end point in unsubscribe, but perhaps if there was some delay it would help smooth over issues with rapid sub/unsub.